File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        start = 0
        maxLength = 1
        n = len(s)

        for i in range(n):
            # Expand around center for odd-length palindromes (aba)
            left, right = i, i
            while left >= 0 and right < n and s[left] == s[right]:
                length = right - left + 1
                if length > maxLength:  # is next longest palindrome
                    maxLength = length
                    start = left
                left -= 1
                right += 1

            # Expand around center for even-length palindromes (abba)
            left, right = i, i + 1
            while left >= 0 and right < n and s[left] == s[right]:
                length = right - left + 1
                if length > maxLength:  # is next longest palindrome
                    maxLength = length
                    start = left
                left -= 1
                right += 1

        return s[start:start + maxLength]

File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.java
class Solution {
    public String longestPalindrome(String s) {
        if (s == null || s.isEmpty()) 
            return "";

        int start = 0; 
        int maxLength = 1; 
        int n = s.length();

        for (int i = 0; i < n; i++) {
            int left = i;
            int right = i;

            // Expand around center for odd-length palindromes
            while (left >= 0 && right < n && s.charAt(left) == s.charAt(right)) {
                int length = right - left + 1;
                if (length > maxLength) {
                    maxLength = length;
                    start = left;
                }
                left--;
                right++;
            }

            left = i;
            right = i + 1;

            // Expand around center for even-length palindromes
            while (left >= 0 && right < n && s.charAt(left) == s.charAt(right)) {
                int length = right - left + 1;
                if (length > maxLength) {
                    maxLength = length;
                    start = left;
                }
                left--;
                right++;
            }
        }

        return s.substring(start, start + maxLength);
    }
}